### PR TITLE
updated option parameters for 1.10

### PR DESCRIPTION
**Pull request by doganmeh**
Commits f7b2ca6 and 6eb17f6 updates the option parameters to version 1.10 (modern ones). 

There is a kind of test (4 total) that fails.. could not follow through.

### CODE REVIEW

**Comment by doganmeh** [comment]
I just added 5de46a0 which fixed documentation for example_app/SatelliteDatatableView

**Comment by doganmeh** [comment]
The four tests failed because of 'ordering' attribute of the datatable class did not work. I just checked the demo at appspot, it does not work there either. And, I am trying to fix it. Here is one line of code that I don't understand, there seems to be some magic here: 

deepcopy1

It seems to be deepcopy'ing something but also introduces new attributes to the self (i.e., .config), and change some existing (i.e., self._meta.ordering)..

Any idea @tiliv ? Thanks in advance..

**Comment by tiliv** [comment]
The reason for cloning the list dates back to legacy config style and keeping things thread-safe when modifying the columns list (because the options and columns were declared on the class itself).  I can't offer insight about the failures just yet though.

I will take a look at this and a few other things tonight!

**Comment by doganmeh** [comment]
Perfect! (for the tonight part :)

**Comment by doganmeh** [comment]
Commit 57f5d22 fixed the ordering bug.. It might be a monkey patch, but might help.. I will put note on the commit as to what was the issue with each part.

File: datatableview/datatables.py [hunk 1]
```diff
@@ -326,8 +326,11 @@ def normalize_config_ordering(self, config, query_config):
 
             ordering.append('%s%s' % (sort_modifier, column.name))
 
-        if not ordering and config['model']:
-            return config['model']._meta.ordering
+        if not ordering:
```
Comment (doganmeh) [inline datatableview/datatables.py]: if ordering not in query string, use Datatable class', and if that is not valid either, use model's.

File: datatableview/datatables.py [hunk 2]
```diff
@@ -326,8 +326,11 @@ def normalize_config_ordering(self, config, query_config):
...
+            if self._meta.ordering:
+                return self._meta.ordering
+            if config['model'] and config['model']._meta.ordering:
+                return config['model']._meta.ordering
         return ordering
```
Comment (doganmeh) [inline datatableview/datatables.py]: if ordering not in query string, use Datatable class', and if that is not valid either, use model's.

File: datatableview/datatables.py [hunk 3]
```diff
@@ -326,8 +326,11 @@ def normalize_config_ordering(self, config, query_config):
 
             ordering.append('%s%s' % (sort_modifier, column.name))
 
-        if not ordering and config['model']:
-            return config['model']._meta.ordering
+        if not ordering:
+            if self._meta.ordering:
+                return self._meta.ordering
+            if config['model'] and config['model']._meta.ordering:
+                return config['model']._meta.ordering
```
Comment (doganmeh) [inline datatableview/datatables.py]: if ordering not in query string, use Datatable class', and if that is not valid either, use model's.

File: datatableview/views/base.py [hunk 4]
```diff
@@ -76,7 +76,6 @@ def get_ajax(self, request, *args, **kwargs):
         """ Called in place of normal ``get()`` when accessed via AJAX. """
 
         datatable = self.get_datatable()
-        datatable.configure()
```
Comment (doganmeh) [inline datatableview/views/base.py]: Earlier, configured datatable in ajax but not in first get. That kept 'ordering' from going into the <th>. So I moved this line to below, so datatable configure either way.

File: datatableview/views/base.py [hunk 5]
```diff
@@ -98,12 +97,21 @@ class AutoMeta:
         kwargs = self.get_datatable_kwargs(**kwargs)
         for meta_opt in opts.__dict__:
             if meta_opt in kwargs:
-                setattr(opts, meta_opt, kwargs.pop(meta_opt))
+                if not (meta_opt == 'ordering' and kwargs['ordering'] is None):
+                    setattr(opts, meta_opt, kwargs.pop(meta_opt))
```
Comment (doganmeh) [inline datatableview/views/base.py]: this extra if keeps `ordering` from `meta` from being overridden by an invalid value from the view class. it was needed.

File: datatableview/views/base.py [hunk 6]
```diff
@@ -98,12 +97,21 @@ class AutoMeta:
...
-        return datatable_class(**kwargs)
+
+        datatable =  datatable_class(**kwargs)
+        # OK there is some magic going in the line above, this line "self.columns = copy.deepcopy(self.base_columns)" changes _meta.ordering some how
+        # todo: fix later, this is a temporary fix
+        if 'ordering' in opts.__dict__: # restore ordering
+            datatable._meta.ordering = opts.ordering
+
+        datatable.configure()
+        return datatable
```
Comment (doganmeh) [inline datatableview/views/base.py]: this was quite weird.. comments should explain it.. i just patched it.. :)

**Comment by tiliv** [comment]
I *think* everything in this PR is solved on master now that #147 is merged!

**Comment by doganmeh** [comment]
I will check tomorrow to make sure :)

**Pull request closed without merging**